CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+
+### Run Project Locally
+
+If you want to go the usual route and run the project locally, though:
+
+- [Install PowerShell](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell)
+- [Install Node.js](https://nodejs.org/en/download/)
+- [Install npm](https://npmjs.com)
+- [Fork the project](https://guides.github.com/activities/forking/#fork)
+
+Then in your (PowerShell) terminal:
+
+- Add coc-powershell to your vim's rtp by `set runtimepath^=/path/to/coc-powershell`
+- `cd path/to/your/coc-powershell`
+- `./build.ps1`
+
+### Run Project Within a Docker Container
+
+If you'd like to use a Docker Container to run and test your changes, we have a prebuild DockerFile available in the `tools` directory.
+
+And you should be ready to go!
+
+## Contribute Documentation
+
+Documentation is a super important, critical part of this project. Docs are how we keep track of what we're doing, how, and why. It's how we stay on the same page about our policies. And it's how we tell others everything they need in order to be able to use this project -- or contribute to it. So thank you in advance.
+
+Documentation contributions of any size are welcome! Feel free to file a PR even if you're just rewording a sentence to be more clear, or fixing a spelling mistake!
+
+To contribute documentation:
+
+- [Set up the project](#project-setup).
+- Edit or add any relevant documentation.
+- Make sure your changes are formatted correctly and consistently with the rest of the documentation.
+- Re-read what you wrote, and run a spellchecker on it to make sure you didn't miss anything.
+- In your commit message(s), begin the first line with `docs:`. For example: `docs: Adding a doc contrib section to CONTRIBUTING.md`.
+- Write clear, concise commit message(s) using [conventional-changelog format](https://github.com/conventional-changelog/conventional-changelog-angular/blob/master/convention.md). Documentation commits should use `docs(<component>): <message>`.
+- Go to https://github.com/yatli/coc-powershell/pulls and open a new pull request with your changes.
+- If your PR is connected to an open issue, add a line in your PR's description that says `Fixes: #123`, where `#123` is the number of the issue you're fixing.
+
+## Contribute Code
+
+We like code commits a lot! They're super handy, and they keep the project going and doing the work it needs to do to be useful to others.
+
+Code contributions of just about any size are acceptable!
+
+The main difference between code contributions and documentation contributions is that contributing code requires inclusion of relevant tests for the code being added or changed. Contributions without accompanying tests will be held off until a test is added, unless the maintainers consider the specific tests to be either impossible, or way too much of a burden for such a contribution.
```

Comment:
We don't have the test pipeline yet 😜

Reply:
I know, and I thought about taking it out... but then I thought perhaps leaving it in and encouraging us to build tests from the get go would be good...

🤔 perhaps rewording it to state that tests are encouraged...

And even better would be adding a section on Contributing Tests...

Reply:
I took out the heavy requirement, instead suggesting that they're encouraged. Perhaps when we have a test pipeline we can add in a part about contributing tests.

Reply:
yeah that would be enough for now;